[翻译完成] src/docs/development/data-and-backend/google-apis.md (#889)
cfug/flutter.cn: src/docs/development/data-and-backend/google-apis.md

The [Google APIs package]({{site.pub-pkg}}/googleapis) exposes dozens of Google
services that you can use from Dart projects.

> [Google APIs package]({{site.pub-pkg}}/googleapis) 暴露了许多你可以从 Dart 项目中使用的 Google 服务。

Reviewer: ```suggestion
[Google APIs package]({{site.pub-pkg}}/googleapis) 提供了许多你可以从 Dart 项目中使用的 Google 服务。
```

  [service accounts](https://cloud.google.com/iam/docs/service-accounts) **should
  not** be used directly from a Flutter application. Doing so requires shipping
  service credentials as part of your application, which is not secure. To use
  these APIs, we recommend creating an intermediate service.
  
  那些需要 [服务账号](https://cloud.google.com/iam/docs/service-accounts) 的 API，
  不应该直接使用自 Flutter 应用。直接使用需要将服务证书作为应用程序的一部分，这是不安全的。
  我们推荐创建一个中间服务，来使用这些 API。

<!-- TODO(kevmoo): Add link to public user guide when available. -->

{{site.alert.end}}

## Overview

## 概览

To use Google APIs, follow these steps.

请遵循以下步骤使用 Google API。

1. Pick the desired API

   选择所需的 API

1. Enable the API

   启用 API

1. Authenticate user with the required scopes

   使用所需的作用域对用户进行身份验证

1. Obtain an authenticated HTTP client

   获取身份验证后的 HTTP 客户端

1. Create and use the desired API class

   创建并使用所需的 API 类

## 1. Pick the desired API

## 1. 选择所需的 API

The documentation for [package:googleapis]({{site.pub-api}}/googleapis) lists
each API as a separate Dart library – in a `name.version` format. Let's look at
[`youtube.v3`]({{site.pub-api}}/googleapis/latest/youtube.v3/youtube.v3-library.html)
as an example.

> 文档 [package:googleapis]({{site.pub-api}}/googleapis) 列举了每一个可以单独作为 Dart 库的 API 
> ——采用 `name.version` 的形式。让我们以 [`youtube.v3`]({{site.pub-api}}/googleapis/latest/youtube.v3/youtube.v3-library.html) 为例。

Reviewer 2: ```suggestion
文档 [package:googleapis]({{site.pub-api}}/googleapis) 采用 `name.version` 的形式，列举了每一个可以单独作为 Dart 库的 API 
。一起看看 [`youtube.v3`]({{site.pub-api}}/googleapis/latest/youtube.v3/youtube.v3-library.html) 这个例子。
```

## 2. Enable the API

> ## 2. 启用API

Reviewer 3: ```suggestion
## 2. 启用 API
```

To use Google APIs you must have a Google account and a Google project. You also
need to enable your desired API.

> 使用 Google API，你必须有一个 Google 账户和一个 Google 项目。你还需要开启所需的 API。

Reviewer 2: ```suggestion
使用 Google API，你必须有一个 Google 账户和一个 Google 项目。你还需要启用所需的 API 服务。
```
我看 Google 的网站里面用的是「启用」，这里 API 我觉得是指的这项服务

In this example, you'd enable
[YouTube Data API v3](https://console.cloud.google.com/apis/api/youtube.googleapis.com).

> 在这个例子中，你开启了 [YouTube Data API v3](https://console.cloud.google.com/apis/api/youtube.googleapis.com)。

Reviewer 2: ```suggestion
在这个例子中，你启用了 [YouTube Data API v3](https://console.cloud.google.com/apis/api/youtube.googleapis.com) 服务。
```
同上

Reviewer: you'd enable 应该是 you would enable，可能翻译的更书面语一些，比如：
在本样例中，你将需要启用 xxxx 服务。

This [`Client`]({{site.pub-api}}/http/latest/http/Client-class.html) instance
includes the nessesary credentials when invoking Google API classes.

> 这个 [`Client`]({{site.pub-api}}/http/latest/http/Client-class.html) 实例

Reviewer: ```suggestion
[`Client`]({{site.pub-api}}/http/latest/http/Client-class.html) 实例
```
「这个」翻译的有点多余。

## More information

## 更多信息

- The
  [`extension_google_sign_in_as_googleapis_auth` example]({{site.pub-pkg}}/extension_google_sign_in_as_googleapis_auth/example)
  is a working implementation of the concepts described on this page.
  

>   这个 [`extension_google_sign_in_as_googleapis_auth` 例子]({{site.pub-pkg}}/extension_google_sign_in_as_googleapis_auth/example) 

Reviewer 2: ```suggestion
  这个例子 [`extension_google_sign_in_as_googleapis_auth` ]({{site.pub-pkg}}/extension_google_sign_in_as_googleapis_auth/example) 
```
[]里面的内容一半翻译一半不翻感觉怪怪的，可以再看看

Author: ```suggestion
  示例 [`extension_google_sign_in_as_googleapis_auth`]({{site.pub-pkg}}/extension_google_sign_in_as_googleapis_auth/example) 
  是本页面所述概念的一个可行的实现。
```
感觉“这个”也是多余的，可以不加。“例子”改成“示例”、“样例”是不是更加正式一点？
